### PR TITLE
Fix openshift/network/third-party suite returning 0 tests.

### PR DESCRIPTION
**Pull request by maheshkurund**
## Description

The suite qualifier references [Suite:k8s] which is no longer appended to test names after the migration to the OpenShift Tests Extension (OTE) framework. The k8s-tests-ext binary does not add [Suite:k8s] to test names, causing the qualifier to match zero tests and breaking CNI certification workflows.

## What

Remove stale `[Suite:k8s]` qualifier from the `openshift/network/third-party` suite definition that causes 0 tests to be selected after the OTE migration.

## Why

After the OTE migration, Kubernetes tests are provided by the `k8s-tests-ext` binary which does not append `[Suite:k8s]` to test names. The qualifier `name.contains("[Suite:k8s]")` matches zero tests, making `openshift-tests run openshift/network/third-party` return "no tests to run".

This breaks the CNI certification workflow documented at [Red Hat CNI Certification §34.6](https://docs.redhat.com/en/documentation/red_hat_software_certification/2026/html/red_hat_software_certification_workflow_guide/con_cni-certification_openshift-sw-cert-workflow-working-with-cloud-native-network-function#running-the-cni-tests_openshift-sw-cert-workflow-working-with-container-network-interface).

## Evidence

```
INFO[0047] Discovered 8265 total tests
INFO[0047] Applying filter: suite-qualifiers             before=8265
INFO[0049] Filter suite-qualifiers completed - removed 8265 tests  after=0 before=8265 removed=8265
Suite run returned error: no tests to run
```

## Fix
Remove the stale [Suite:k8s] requirement from the qualifier so that sig-network conformance and NetworkPolicy tests are correctly selected regardless of which binary provides them.

## Summary by CodeRabbit

* **Tests**
  * Updated network conformance test suite selection criteria to refine the test filtering logic for network-related tests.

### CODE REVIEW

File: pkg/testsuites/standard_suites.go [hunk 1]
```diff
@@ -317,7 +317,7 @@ var staticSuites = []ginkgo.TestSuite{
 		The conformance testing suite for certified third-party CNI plugins.
 		`),
 		Qualifiers: []string{
-			`name.contains("[Suite:k8s]") && name.contains("[sig-network]") && 
+			`name.contains("[sig-network]") &&
```
Comment (danwinship) [inline pkg/testsuites/standard_suites.go:320]: This will cause the suite to match tests from origin itself that are labeled `[sig-network]`, which we did not want to include. The `openshift/network/third-party` suite is (at least currently) supposed to be limited only to upstream tests.

Comment (maheshkurund) [reply pkg/testsuites/standard_suites.go:320]: Thanks for the taking a look. Agree with you, this is not right fix.

Comment (jfrancin) [reply pkg/testsuites/standard_suites.go:320]: @maheshkurund We need a proper fix for this as soon as possible, as _no_ CNI partner can now certify their drivers on v4.21 and later. @danwinship What would be the correct test here?
